yogadl/_core.py

```diff
@@ -76,6 +76,7 @@ def stream(
         shuffle_seed: Optional[int] = None,
         shard_rank: int = 0,
         number_of_shards: int = 1,
```

Comment:
I know this isn't part of this PR, but can we rename this `num_shards`?  `number_of_shards` is much longer and adds no extra clarity.

Reply:
Done